extensions/amp-script/0.1/amp-script.js

```diff
@@ -640,7 +649,8 @@ export class SanitizerImpl {
             dev().error(TAG, 'Invalid AMP.setState() argument: %s', value);
           });
           if (state) {
-            bind.setState(state, /* skipEval */ true, /* skipAmpState */ false);
+            const skipEval = !this.userActivationTracker_.isActive();
+            bind.setState(state, skipEval, /* skipAmpState */ false);
```

Comment:
If skipEval == true, let's `user().warn` that the `AMP.setState` only updated the page state since there was no backing user gesture.

Reply:
Done. Thanks for the improvement!